src/vnet/devices/memif/device.c

```diff
@@ -129,66 +129,80 @@ memif_interface_tx_inline (vlib_main_t * vm, vlib_node_runtime_t * node,
   else
     free_slots = ring_size - head + tail;
 
-  while (n_left > 5 && free_slots > 1)
+  while (n_left > 3 && free_slots > 3)
     {
-      if (PREDICT_TRUE (head + 5 < ring_size))
+      if (PREDICT_TRUE (head + 3 < ring_size))
 	{
 	  CLIB_PREFETCH (memif_get_buffer (mif, ring, head + 2),
 			 CLIB_CACHE_LINE_BYTES, STORE);
 	  CLIB_PREFETCH (memif_get_buffer (mif, ring, head + 3),
 			 CLIB_CACHE_LINE_BYTES, STORE);
-	  CLIB_PREFETCH (&ring->desc[head + 4], CLIB_CACHE_LINE_BYTES, STORE);
-	  CLIB_PREFETCH (&ring->desc[head + 5], CLIB_CACHE_LINE_BYTES, STORE);
+	  CLIB_PREFETCH (&ring->desc[head + 2], CLIB_CACHE_LINE_BYTES, STORE);
+	  CLIB_PREFETCH (&ring->desc[head + 3], CLIB_CACHE_LINE_BYTES, STORE);
```

Comment:
I intentionally prefetched 2 iterations in advance. Please leave it as it is. Later we will rerun perf test...